src/path-filter.ts

```diff
@@ -3,37 +3,41 @@ import * as isGlob from 'is-glob';
 import * as micromatch from 'micromatch';
 import * as url from 'url';
 import { ERRORS } from './errors';
+import { getUrl } from './url';
 
-export function matchPathFilter(pathFilter: Filter = '/', uri: string, req: Request): boolean {
-  // single path
-  if (isStringPath(pathFilter as string)) {
-    return matchSingleStringPath(pathFilter as string, uri);
-  }
-
-  // single glob path
-  if (isGlobPath(pathFilter as string)) {
-    return matchSingleGlobPath(pathFilter as unknown as string[], uri);
-  }
-
-  // multi path
-  if (Array.isArray(pathFilter)) {
-    if (pathFilter.every(isStringPath)) {
-      return matchMultiPath(pathFilter, uri);
-    }
-    if (pathFilter.every(isGlobPath)) {
-      return matchMultiGlobPath(pathFilter as string[], uri);
-    }
-
-    throw new Error(ERRORS.ERR_CONTEXT_MATCHER_INVALID_ARRAY);
-  }
+/**
+ * Express mutates `Request::url`, demanding special case
+ * handling for this single field.
+ * {@link https://github.com/expressjs/express/issues/4854}
+ */
 
-  // custom matching
-  if (typeof pathFilter === 'function') {
-    const pathname = getUrlPathName(uri);
-    return pathFilter(pathname, req);
+export function matchPathFilter(pathFilter: Filter = '/', req: Request): boolean {
```

Comment:
this is the 2nd biggest change with this PR since we last reviewed. `uri: string` was dropped as a param. now, `url` is always extract from the `req` param, safely.

a nice little bit of simplification 🧹 . this was done as part of the de-express-ificationthe